### ISSUE TITLE
FR: Imgix Support

### ISSUE THREAD

**Issue by goellner** [spatie/statamic-responsive-images#42]
Feature Request: Add support for imgix instead of glide

statamic-responsive-images is just awesome to work with and it would be even better, if we could use imgix instead of glide. Then we have the same responsive image library and the syntax could stay the same.

**Comment by riasvdv** [comment]
I won't add Imgix support within the addon as we don't use the service and it's not a maintenance burden we like to take on. I have however made the job that generates the images configurable, which should allow you to replace it by your own Imgix implementation that generates imgix urls.

**Comment by goellner** [comment]
thank you for making it configurable in the first place! could you give me an example, how i can hook into this generation config? do i have to publish something with artisan? when i get the settings from the fieldset, then it should be just generating a url, where the parameters get mapped so imgix understands it.

**Comment by goellner** [comment]
I managed to get this working, but I have one problem. The url gets stripped somehow in the tag itself. I return the full URL in the custom job, but just the path is used in the tag.

is this something that has to be configurable too? 


```
<?php

namespace App\Jobs;

use Illuminate\Support\Collection;
use Statamic\Contracts\Assets\Asset;
use Statamic\Facades\Asset as AssetFacade;
use Spatie\ResponsiveImages\Jobs\GenerateImageJob;

class GenerateImgixImageJob extends GenerateImageJob
{
  protected function imageUrl(): string
  {
    $imgixUrl = $this->getManipulatedImage($this->asset, collect($this->params));
    return $imgixUrl;
  }

  private function getManipulatedImage(Asset $asset, Collection $params)
  {
    $src = $asset->url();
    $fit = $params->get('fit', null);
    $assetFacade = AssetFacade::findByUrl($src);
    $focalPoint = "";
    $focalPoint = $assetFacade->data()->get("focus");
    if (isset($fit["value"]) && $fit["value"] === "crop_focal") {
      if (isset($focalPoint)) {
        $focus = explode("-", $focalPoint);
        $focusX = intval($focus[0], 10) / 100;
        $focusY = intval($focus[1], 10) / 100;
        $focalPoint = '&fit=crop&crop=focalpoint&fp-x=' . $focusX . '&fp-x=' . $focusY;
      } else {
        $focalPoint = "&fit=crop";
      }
    } else {
      $focalPoint = "&fit=crop";
    }

    $width = $params->get('width', null) ? "w=" . $params->get('width', null) : "";
    $height = $params->get('height', null) ? "&h=" . $params->get('height', null) : "";

    $ret = $src . "?" . $width . $height . $focalPoint;

    return $ret;
  }
}
```

**Comment by goellner** [comment]
I found the culprit, it is on this line: https://github.com/spatie/statamic-responsive-images/blob/28dc6a05c4d143a75e3dbd83e7a5c46ef2c98d24/src/Breakpoint.php#L103

Could it be a  setting in the plugin by any chance @riasvdv? When it returns the full url, then everything should be working

**Comment by riasvdv** [comment]
I'll move that function to the job, that should fix it for you

**Comment by riasvdv** [comment]
Should be fixed for you in 2.4.3

**Comment by goellner** [comment]
Thank you! Working like a charm
